src/Lykke.Service.Assets.Repositories/AssetConditionLayerRepository.cs

```diff
@@ -11,148 +12,84 @@ namespace Lykke.Service.Assets.Repositories
 
-        public async Task InsetLayerAsync(IAssetConditionLayer layer)
-        {
-            var entity = new AssetConditionLayerEntity(
-                GetAssetConditionLayerPartitionKey(),
-                GetAssetConditionLayerRowKey(layer.Id.Trim().ToLower()),
-                layer.Priority,
-                layer.Description,
-                layer.ClientsCanCashInViaBankCards,
-                layer.SwiftDepositEnabled);
-
-            await _assetConditionLayerTable.InsertAsync(entity);
-
-            if (layer.AssetConditions != null)
-            {
-                foreach (var assetCondition in layer.AssetConditions.Values)
-                {
-                    await InsertOrUpdateAssetConditionsToLayerAsync(layer.Id, assetCondition);
-                }
-            }
+            return result;
         }
 
-        public async Task UpdateLayerAsync(IAssetConditionLayer layer)
+        /// <summary>
+        /// Inserts a new entity if it does not exist, otherwise updates existing entity.
+        /// </summary>
+        /// <param name="layer">The asset conditon layer.</param>
+        /// <returns></returns>
+        public async Task InsertOrUpdateAsync(IAssetConditionLayer layer)
         {
-            await _assetConditionLayerTable.ReplaceAsync(GetAssetConditionLayerPartitionKey(),
-                GetAssetConditionLayerRowKey(layer.Id),
-                current => current.Apply(layer.Priority, layer.Description, layer.ClientsCanCashInViaBankCards,
-                    layer.SwiftDepositEnabled));
-        }
+            var entity = new AssetConditionLayerEntity(GetPartitionKey(), GetRowKey(layer.Id));
 
-        public async Task DeleteLayerAsync(string layerId)
-        {
-            var assetConditions = await _assetConditionTable.GetDataAsync(GetAssetConditionPartitionKey(layerId));
-            await _assetConditionTable.DeleteAsync(assetConditions);
+            Mapper.Map(entity, layer);
 
-            await _assetConditionLayerTable.DeleteAsync(GetAssetConditionLayerPartitionKey(),
-                GetAssetConditionLayerRowKey(layerId));
-        }
-        
-        private static string GetAssetConditionLayerPartitionKey()
-        {
-            return "ConditionLayer";
+            await _storage.InsertOrReplaceAsync(entity);
         }
 
-        private static string GetAssetConditionLayerRowKey(string layerId)
+        /// <summary>
+        /// Deletes an asset conditon layer.
+        /// </summary>
+        /// <param name="layerId">The layer id.</param>
+        /// <returns></returns>
+        public async Task DeleteAsync(string layerId)
         {
-            return layerId;
+            await _storage.DeleteAsync(GetPartitionKey(), GetRowKey(layerId));
         }
 
-        private static string GetAssetConditionPartitionKey(string layerId)
-        {
-            return $"Condition_{layerId}";
-        }
+        private static string GetPartitionKey()
+            => "ConditionLayer";
```

Comment:
Use .CalculateHexHash32(n) for better partitioning

Reply:
This partition key is used in production environment. In this case it will be small partition around 10-30 rows.